// gitbook-plugin-mathjax/_assets/plugin.js
(function(f){if(typeof exports==="object"&&typeof module!=="undefined"){module.exports=f()}else if(typeof define==="function"&&define.amd){define([],f)}else{var g;if(typeof window!=="undefined"){g=window}else if(typeof global!=="undefined"){g=global}else if(typeof self!=="undefined"){g=self}else{g=this}g.GitBookPlugin = f()}})(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
"use strict";

var GitBook = require('gitbook-core');
var React = GitBook.React;

/**
 * Math block when using SVG mode.
 * @type {ReactClass}
 */

var MathJaxSVG = React.createClass({
    displayName: "MathJaxSVG",

    propTypes: {
        inline: React.PropTypes.bool.isRequired,
        filename: React.PropTypes.string.isRequired
    },

    render: function render() {
        var _props = this.props,
            inline = _props.inline,
            filename = _props.filename;


        var img = React.createElement(GitBook.Image, { src: filename });

        if (inline) {
            return img;
        } else {
            return React.createElement(
                "div",
                { className: "MathJaxBlock-MathJaxSVG" },
                img
            );
        }
    }
});

/**
 * Math templating block.
 * @type {ReactClass}
 */
var MathJaxBlock = React.createClass({
    displayName: "MathJaxBlock",

    propTypes: {
        isSVG: React.PropTypes.bool.isRequired,
        inline: React.PropTypes.bool,
        content: React.PropTypes.string,
        filename: React.PropTypes.string
    },

    render: function render() {
        var _props2 = this.props,
            isSVG = _props2.isSVG,
            inline = _props2.inline,
            content = _props2.content,
            filename = _props2.filename;


        return React.createElement(
            "span",
            { className: "MathJaxBlock" },
            React.createElement(GitBook.ImportCSS, { href: "gitbook/mathjax/mathjax.css" }),
            isSVG ? null : React.createElement(GitBook.ImportScript, { src: "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" }),
            isSVG ? React.createElement(MathJaxSVG, { filename: filename, inline: inline }) : React.createElement(
                "span",
                null,
                React.createElement(
                    "script",
                    { type: "math/tex; " + (inline ? '' : 'mode=display') },
                    content
                )
            )
        );
    }
});

module.exports = GitBook.createPlugin({
    activate: function activate(dispatch, getState, _ref) {
        var Components = _ref.Components;

        dispatch(Components.registerComponent(MathJaxBlock, { role: 'block:math' }));
    }
});

},{"gitbook-core":"gitbook-core"}]},{},[1])(1)
});
